duckduckgo/iOS
# Add Sync e2e test flows

Task/Issue URL: https://app.asana.com/0/0/1205914590479611/f
Tech Design URL:
CC:

**Description**:


This PR adds the critical e2e tests identified in the task above. 

Most of them will need an update soon as the new sync setup flow is implemented, but the most interesting part of this PR is that it sets up the integration with the JS test script, so it's still worth it to merge it.

**Steps to test this PR**:
1. Verify CI is green.

To test locally:
2. Download https://github.com/duckduckgo/sync_crypto (You want to use [this](loremattei/make-run-locally) branch if the PR has not been merged yet).
3. Run `node index.js` and copy the recovery code that the script outputs.
4. Open an iOS simulator and install the iOS app. 
5. Run `maestro test -e CODE=<Paste the code here> ./.maestro/sync_tests`.
6. When finished, delete the test sync account (The easiest way to do it is using the dedicated option in the iOS app).




---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
